### PR TITLE
SnakeKeyListener and tests implemented

### PR DESCRIPTION
**Pull request by lliebman**

### CODE REVIEW

File: src/test/java/touro/snake/SnakeKeyListenerTest.java [hunk 1]
```diff
@@ -2,27 +2,84 @@
...
+        //given
+        Snake snake = mock(Snake.class);
+        SnakeKeyListener snakeKeyListener = new SnakeKeyListener(snake);
+        KeyEvent keyEvent = mock(KeyEvent.class);
+
+        //when
+        snakeKeyListener.keyPressed(keyEvent);
+
+        //then
+        verifyZeroInteractions(snake);
```
Comment (ester-agishtein) [inline src/test/java/touro/snake/SnakeKeyListenerTest.java:83]: What is this command/what does this do? Just curious.

Comment (lliebman) [reply src/test/java/touro/snake/SnakeKeyListenerTest.java:83]: It checks to see if snake is ever called during the test. If it is never called/has zero interactions made with it, then it's happy. In the case of a nonArrow key being pressed, I'm testing to make sure that nothing will happen.

File: src/main/java/touro/snake/SnakeKeyListener.java [hunk 2]
```diff
@@ -18,6 +18,18 @@ public SnakeKeyListener(Snake snake) {
     public void keyPressed(KeyEvent e) {
         super.keyPressed(e);
```
Comment (lliebman) [inline src/main/java/touro/snake/SnakeKeyListener.java:19]: @andrewoid I'm curious about this line. Why do we need to call the default implementation of the keystroke?

File: src/test/java/touro/snake/SnakeKeyListenerTest.java [hunk 3]
```diff
@@ -2,27 +2,84 @@
...
+
+        //then
+        verify(snake).turnTo(Direction.West);
     }
 
     @Test
     public void keyPressed_right() {
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        //given
+        Snake snake = mock(Snake.class);
```
Comment (dnagelCS) [inline src/test/java/touro/snake/SnakeKeyListenerTest.java:30]: How does mocking a class work when a class takes in a parameter? For instance Snake has snakeHeadStateMachine as its parameter.

File: src/main/java/touro/snake/SnakeKeyListener.java (hunk 2 above)
Comment (andrewoid) [reply src/main/java/touro/snake/SnakeKeyListener.java:19]: It's put in automatically.
`KeyAdapter` is an abstract class that could have an implementation of `keyPressed` that needs to run as well. So it's always a good idea in these situations to call the super class's implementation when overriding a method.

File: src/test/java/touro/snake/SnakeKeyListenerTest.java (hunk 3 above)
Comment (andrewoid) [reply src/test/java/touro/snake/SnakeKeyListenerTest.java:30]: You don't need to mock the parameters of the class you are mocking. 
The mock looks like the class you're mocking, but it has no effect, meaning it doesn't need it's parameters.

File: src/main/java/touro/snake/SnakeKeyListener.java [hunk 4]
```diff
@@ -17,7 +17,18 @@ public SnakeKeyListener(Snake snake) {
     @Override
     public void keyPressed(KeyEvent e) {
         super.keyPressed(e);
-        // Liebman
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        switch(e.getKeyCode()) {
+            case KeyEvent.VK_LEFT :
+                snake.turnTo(Direction.West);
+
```
Comment (andrewoid) [inline src/main/java/touro/snake/SnakeKeyListener.java:23]: You are missing break statements

File: src/test/java/touro/snake/SnakeKeyListenerTest.java [hunk 5]
```diff
@@ -2,27 +2,84 @@
...
     @Test
     public void keyPressed_left() {
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        //given
+        Snake snake = mock(Snake.class);
+        SnakeKeyListener snakeKeyListener = new SnakeKeyListener(snake);
+        KeyEvent keyEvent = mock(KeyEvent.class);
+
+        //when
+        when(keyEvent.getKeyCode()).thenReturn(KeyEvent.VK_LEFT);
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeKeyListenerTest.java:20]: move the when(), into the given.

File: src/test/java/touro/snake/SnakeKeyListenerTest.java [hunk 6]
```diff
@@ -2,27 +2,84 @@
...
+        snakeKeyListener.keyPressed(keyEvent);
+
+        //then
+        verify(snake).turnTo(Direction.West);
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeKeyListenerTest.java:24]: add verifyNoMoreInteractions(snake)

**Pull request merged**